gammapy/maps/wcs/ndmap.py

```diff
@@ -650,16 +652,22 @@ def to_region_nd_map(
             # Casting needed as interp_by_coord transforms boolean
             data = data.astype(self.data.dtype)
         else:
-            cutout, mask = self.cutout_and_mask_region(region=region)
+            cutout, cutout_mask = self.cutout_and_mask_region(region=region)
+            mask = cutout_mask.interp_to_geom(cutout.geom)
```

Comment:
does this return a boolean ?
for mask I think we should use `method="nearest"`.